src/worldstate/routine/Gate/src/alignWithGate.cpp

```diff
@@ -0,0 +1,29 @@
+//
+// Created by joel on 12/05/18.
+//
+#include "alignWithGate.h"
+
+void alignWithGate::setupNodeSubscriptions(ros::NodeHandle nh) {
+    std::string gateDetectTopic = "/gateDetect/output";
+    nh.subscribe(gateDetectTopic, 10, &alignWithGate::gateDetectCallBack, this);
+}
+
+void alignWithGate::gateDetectCallBack(
+const gate_detect::gateDetectMsg::ConstPtr& msg) {
+    worldstate::state_msg msg_to_publish;
+    msg_to_publish.state = worldstate::state_msg::aligningWithGate;
+
+    if (!(msg->detectLeft || msg->detectRight || msg->detectTop))
```

Comment:
I think that this would be a bit easier to read (and logically equivalent) if we wrote it as `if (!msg->detectLeft && !msg->detectRight && !msg->detectTop)`